build_defs/test_helpers.bzl

```diff
@@ -82,3 +89,68 @@ def extracted_datalog_string_test(
             "@souffle//:souffle_include_lib",
         ],
     )
+
+def run_taint_exec_compare_check_results(
+        name,
+        check_file,
+        other_input_files,
+        visibility = None):
+    """Run the Souffle-generated taint executable and ensure that the failing
+    check output file is empty.
+
+    Args:
+      name: String; Name of the test.
+      input_files: List; List of labels indicatinginput files to taint_exec.
+    """
+
+    input_files = other_input_files + [check_file]
+
+    facts_dir_opts = [
+        "--facts=`dirname $(location {})`".format(input_file) for input_file in input_files
+    ]
+
+    native.genrule(
```

Comment:
A single line comment for each genrule would be useful as documentation.

Reply:
Will do.

Reply:
Done.